### ISSUE TITLE
Next steps

### ISSUE THREAD

**Issue by zeileis** [cran-task-views/ctv#7]
Now that the R package `ctv` version 0.9-0 has been [released on CRAN](https://CRAN.R-project.org/package=ctv) and also got its own [pkgdown website](https://ctv.R-Forge.R-project.org/) we can move forward with the next steps. If you have any comments/concerns, please let me know in the comments.

1. _Step:_ Make the `ctv` and `Econometrics` repositories public.  
   _Who:_ @zeileis  
   _Timeline:_ Before we notify the task view maintainers. Ideally tomorrow.  
   _Question:_ I think all discussions we had in the issue are ok to go public. Would anyone prefer not making some of the issues public?
2. _Step:_ Notify the task view maintainers about the upcoming steps (XML -> R/Markdown, R-Forge -> GitHub, single maintainers -> teams). Ask for GitHub handles, needed both for correct mapping in svn -> git conversion and for permissions in GitHub repositories.  
   _Who:_ @zeileis  
   _Timeline:_ Ideally tomorrow (2021-12-16).  
   _Question:_ Anything else we need to mention to the maintainers at this stage?
3. _Step:_ Delete the current `ctv-from-svn` repository and re-import from R-Forge. After that the .ctv files on R-Forge should not be edited further (or only in sync with the .md files on GitHub).  
   _Who:_ @eddelbuettel  
   _Timeline:_ Decided by Dirk. My suggestion would be that all changes on R-Forge should be completed by 2021-12-22 23:59:99 UTC.  
   _Question:_ Would this work for you, Dirk? Otherwise feel free to set another timeline.
4. _Step:_ Set up a GitHub repository for each task view, get the .ctv file for the migration (including history), convert to .md, add README.md (similar to Econometrics task view).  
   _Who:_ @eddelbuettel for the first copying and @zeileis for the conversion and README?  
   _Timeline:_ Some point between Step 3 and 2022-01-09.  
   _Question:_ Dirk, is this ok for you? Or should we try to distribute the work a bit more?
5. _Step:_ "Onboarding" of all task view maintainers, helping them with the new format, incorporating co-maintainers, and potentially other improvement.  
   _Who:_ All of us, every one for six or seven task views.  
   _Timeline:_ After Step 4 and end of January.  
   _Question:_ What to do with "orphaned" task views?
6. _Step:_ Adapt scripts on CRAN to fetch task views from GitHub rather than R-Forge.  
   _Who:_ @zeileis and Kurt Hornik.  
   _Timeline:_ After Step 4.  
   _Question:_ What is the easiest way to get copies of the .md files from GitHub, ideally without the need for a token?

**Comment by zeileis** [comment]
Step 1 is done now, `ctv` and `Econometrics` are public, the former being pinned.

**Comment by rociojoo** [comment]
Thanks for all the great work! Just chiming in to say that I'd prefer to move the public github repo of the `Tracking` ctv to cran-task-views myself once I update everything in January.

**Comment by zeileis** [comment]
In case of the Tracking view I think it would make more sense to use the version from R-Forge. We lose the pre-history from before the view was released on CRAN but we have a much more granular history of package removal due to CRAN retirement. So far there were also no PRs from non-maintainers that one could argue should be preserved. Hence, I feel it's cleaner to transfer/convert what we have in the SVN and then do your changes on top of it afterwards.

**Comment by eddelbuettel** [comment]
I plan to unconditionally transfer _all_ views from SVN into _private_ repos at GH.  You call can then decide one-by-one which to keep, which to toss, which to edit, ...  

The past of the project is in SVN.  So I am with @zeileis that a good default is to stick with the SVN version unless some truly meaningful arguments are presented that make sticking with the github history preferable.  Ultimately all @zeileis's call.

**Comment by rociojoo** [comment]
Got it. It sounds good then. I'll keep my public repo, keep working then, and update from there to the repo here (once it is created).

**Comment by zeileis** [comment]
I think it's time that I post some information about the progress regarding this issue. Except for a couple of repositories with special requirements we had finished Steps 1-4 for all task views by 2021-12-31.

As for Step 5 I have opened more or less the same issue asking for general improvements in all of the task view repositories. Where applicable I also asked for adding co-maintainers and potentially added further view-specific aspects that I noticed. A good share of the task view maintainers became active following this but many so far just accepted my invitation to become admins of their repositories. Very few have actually completed the tasks and closed the issue.

Therefore, I plan to send out another e-mail to all task view maintainers early next week to remind them about it. The aim would be that all maintainers at least had a look at the new .md files and confirmed that this is ok for update on CRAN - by end of the month. Then we could make the repositories public and update CRAN accordingly.

For now I think I can still coordinate this myself because I was the one in contact with the maintainers over the last years. But I may ask for some specific support from you later this month. Also, if you are familiar with some of the task views yourself and have specific suggestions, feel free to join in the discussions of the issues in the respective repositories!

**Comment by zeileis** [comment]
Finally, some update again. I have worked with several of the authors to make their GitHub projects public and <https://github.com/cran-task-views/ctv/> currently lists nine (of 38) task views so that they can be fetched by the new scripts on CRAN. But this also means that we still have a long way to go before we have completed the transition. I just sent out an e-mail to all task view maintainers, informing them about the next steps. To monitor the progress following this e-mail and to chase the irresponsive maintainers, I suggest that we split up the task views between us in the following way.

* David @davidjohannesmeyer: Bayesian, GraphicalModels, NaturalLanguageProcessing

* Dirk @eddelbuettel: Distributions, ExtremeValue, Robust, TimeSeries

* Nathalie @tuxette: MedicalImaging, Pharmacokinetics, Phylogenetics, Survival

* Rocio @rociojoo: DifferentialEquations, ExperimentalDesign, FunctionalData, ReproducibleResearch

* Roger @rsbivand: ChemPhys, Environmetrics, TeachingStatistics

* Achim @zeileis: Cluster, MachineLearning, MetaAnalysis, NumericalMathematics, Optimization, Psychometrics, WebTechnologies

First, please check whether the maintainers have accepted the invitation to join the GitHub project and have a look at whether there was any follow-up in the "Issues" or the "Code" itself.

Second, if there is no follow-up from the maintainers by Monday (Mar 7), then please follow up in the issues of that project and @ the maintainer (and all co-maintainers, if any) explicitly.

Third, when the issues have been addressed satisfactorily please turn the GitHub project "Public" and list it in https://github.com/cran-task-views/ctv/blob/main/README.md

Thanks!

**Comment by zeileis** [comment]
One addition that I forgot to mention: The scripts on CRAN (Step 6) are ready as well. So we "just" need to bring all task views into shape as listed above. If there are no objections to the general plan, I'll close this issue and set up a separate task list for us. If you agree, a quick "thumbs up" reaction is enough.

**Comment by rsbivand** [comment]
TeachingStatistics now public.
Environmetrics: Some movement 2022-03-02. Environmetrics.md builds and checks OK.
ChemPhys: General improvement issue closed today by k-m-m. ChemPhys.md builds and checks OK.

Spatial may be made public when you are ready.

**Comment by tuxette** [comment]
MedicalImaging: no activity (the maintainer has accepted the invitation)
Pharmacokinetics: only two commits (in the past two days); checks OK
Phylogenetics: no activity (the maintainer has not accepted the invitation)
Survival: no activity (one of the two maintainers has accepted the invitation; the other has not been invited yet?)

I'll contact all of them on Monday but it sounds like I'll have a hard time ;)

**Comment by rociojoo** [comment]
* DifferentialEquations: Two people are listed as maintainers. No activity. The contact email corresponds to one of them and the other one received and accepted the invitation. Achim @zeileis , do you know why the contact email and github account are not the same person's?
* ExperimentalDesign: Two people invited and accepted the invitation. No activity.
* FunctionalData: One person invited and accepted the invitation. No activity. 
* ReproducibleResearch: Six people invited and 4 accepted the invitation. Commits made by jdblishchak in early February and a message "The Task View needs an overhaul given the dominance of R Markdown. The beginning of this year has been off to too hectic of a start for me to tackle this yet. But I don't think this should prevent the deployment of the new infrastructure to CRAN. The current version still fulfills the goal of curating many useful CRAN packages for reproducible research."

I'll contact all of them on Monday or Tuesday.

**Comment by tuxette** [comment]
Updates:
- MedicalImaging: has been updated yesterday and Achim was faster than me to answer so I consider this repo as active at the moment and wait for the maintainer to answer Achim's message (I'll make a reminder within a week or so if nothing has changed: what do you think Achim @zeileis ?)
- [X] Pharmacokinetics: only two commits (in the past two days); checks OK and turned public already
- Phylogenetics: no activity (the maintainer has not accepted the invitation yet): I've seen that you already have sent an email to the 3rd, Achim @zeileis so maybe I'd better wait a few more days for a reminder?
- Survival: no activity (one of the two maintainers has accepted the invitation). I've just sent a message.
Let me know if you think that I should do something else.

**Comment by zeileis** [comment]
Thanks for all the follow-ups. For easier tracking of what still needs to be done, I've opened a separate Issue https://github.com/cran-task-views/ctv/issues/19. Overall I think we are making good progress. In many of the projects that are not public, yet, the maintainers have already started to take action. So we "just" need to shove them a little bit in the right direction, I think.

I'm collecting a little bit of more detailed feedback below. Further progress can then be tracked in the new issue I've opened. I'm closing this one.

* Roger: Thanks for the feedback. Only Environmetrics remains but I would expect some follow-up from the maintainer soon.
* Nathalie: My hope is that MedicalImaging will become ready soon. For Phylogenetics I don't think we need to send further reminders - if there is no update in the next days, we need to treat it as orphaned. Finally, Arthur, the maintainer of the Survival view is usually responsive - so hopefully there will be some follow-up in the next days.
* Rocío: Please post follow-up comments with reminders for ExperimentalDesign and FunctionalData and please @ all co-maintainers explicitly to make sure that they receive the message. For DifferentialEquations and ReproducibleResearch I have already posted follow-up comments in the respective issues.
